src/upload-lib.ts

```diff
@@ -107,19 +112,34 @@ async function uploadPayload(
 
   const client = api.getApiClient();
 
-  const response = await client.request(
-    "PUT /repos/:owner/:repo/code-scanning/analysis",
-    {
-      owner: repositoryNwo.owner,
-      repo: repositoryNwo.repo,
-      data: payload,
-    },
-  );
-
-  logger.debug(`response status: ${response.status}`);
-  logger.info("Successfully uploaded results");
+  try {
+    const response = await client.request(
+      "PUT /repos/:owner/:repo/code-scanning/analysis",
+      {
+        owner: repositoryNwo.owner,
+        repo: repositoryNwo.repo,
+        data: payload,
+      },
+    );
 
-  return response.data.id;
+    logger.debug(`response status: ${response.status}`);
+    logger.info("Successfully uploaded results");
+    return response.data.id;
+  } catch (e) {
+    if (util.isHTTPError(e)) {
+      switch (e.status) {
+        case 403:
+          core.warning(GENERIC_403_MSG);
```

Comment:
```suggestion
          core.warning(e.message || GENERIC_403_MSG);
```
for parity with the prior message emitted in the status report file.